app/models/order_detail.rb

```diff
@@ -771,6 +747,34 @@ def problem_order?
     problem
   end
 
+  def can_dispute?
+    in_review? && dispute_at.blank?
+  end
+
+  def in_dispute?
+    dispute_at.present? && dispute_resolved_at.nil? && !canceled?
+  end
+
+  def in_review?
+    reviewed_at.try(:future?) && !in_dispute?
+  end
+
+  def reviewed?
+    reviewed_at.try(:past?) && !in_dispute?
+  end
```

Comment:
Yeah, independent makes sense, I was just trying to make sure I got it right in my head. `past?` and `future?` are the right calls, I just got tangled comparing the old with the new..